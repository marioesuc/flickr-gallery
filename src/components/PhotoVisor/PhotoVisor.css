.PhotoVisor-container {
  position: fixed;
  top: 0px;
  margin: 0px;
  height: 100vh;
  width: 100vw;
  background: rgba(0, 0, 0, 0.7);
  z-index: 5;
}

.PhotoVisor-image {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: auto;
  height: 90%;
  max-width: 90%;
  box-shadow: 0px 5px 15px 2px rgba(0, 0, 0, 0.5);
}

.PhotoVisor-close {
  position: absolute;
  top: 0;
  right: 0;
  color: rgba(255, 255, 255, 0.8);
  font-size: 20px;
  font-weight: bold;
  border: 1px solid rgba(255, 255, 255, 0.8);
  padding: 10px;
  border-radius: 5px;
  margin: 20px;
  transition: 0.4s;
  cursor: pointer;
}

.PhotoVisor-close:hover {
  background: rgba(255, 255, 255, 0.8);
  color: #404040;
  transition: 0.2s;
}
