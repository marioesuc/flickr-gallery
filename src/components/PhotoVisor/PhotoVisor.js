import React from 'react';
import './PhotoVisor.css';

// Custom PhotoVisor component
class PhotoVisor extends React.Component {
  handleOnArrowClick = relativeIndex => this.props.onArrowClick(relativeIndex);
  handleOnVisorClose = () => this.props.onVisorClose();

  render() {
    const { visible, photo } = this.props;

    if (visible) {
      const { source, title } = photo;

      return (
        <div className='PhotoVisor-container'>
          <div className='PhotoVisor-ImageContainer'>
            <img alt={source} className='PhotoVisor-image' src={source} />
            <div className='PhotoVisor-ArrowsContainer'>
              <div
                className='PhotoVisor-LeftArrowContainer'
                onClick={() => this.handleOnArrowClick(-1)}
              >
                ←
              </div>
              <div
                className='PhotoVisor-RightArrowContainer'
                onClick={() => this.handleOnArrowClick(1)}
              >
                →
              </div>
            </div>
            <div className='PhotoVisor-Title'>
              <span>{title}</span>
            </div>
          </div>
          <button
            className='PhotoVisor-close'
            onClick={this.handleOnVisorClose}
          >
            Close
          </button>
          <div
            className='PhotoVisor-background'
            onClick={this.handleOnVisorClose}
          />
        </div>
      );
    }

    return null;
  }
}

export default PhotoVisor;
