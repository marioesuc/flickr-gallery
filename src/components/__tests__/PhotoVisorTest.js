import React from 'react';
import { shallow, configure } from 'enzyme';
import PhotoVisor from '../PhotoVisor/PhotoVisor';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('PhotoVisor', () => {
  it('renders correctly when active', () => {
    const props = {
      visible: true,
      photo: {
        source: 'source'
      },
      onVisorClose: jest.fn()
    };

    const wrapper = shallow(<PhotoVisor {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when inactive', () => {
    const props = {
      visible: false,
      photo: null,
      onVisorClose: jest.fn()
    };

    const wrapper = shallow(<PhotoVisor {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
