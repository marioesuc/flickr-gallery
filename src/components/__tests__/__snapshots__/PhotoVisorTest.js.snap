// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoVisor renders correctly when active 1`] = `ShallowWrapper {}`;

exports[`PhotoVisor renders correctly when inactive 1`] = `ShallowWrapper {}`;
